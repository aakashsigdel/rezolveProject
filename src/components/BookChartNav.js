import React from 'react'
import { Link } from 'react-router'

import './BookChartNav.scss'

const BookChartNav = props =>
  <div className="nav-container">
    <Link to="/" activeClassName="nav-link-active" className="nav-link" onlyActiveOnIndex={true}>
      MY BOOKS
    </Link>
    <div className="nav-element-separator">&nbsp; | &nbsp;</div>
    <Link to="/charts" activeClassName="nav-link-active" className="nav-link">
      CHARTS
    </Link>
  </div>

export default BookChartNav
