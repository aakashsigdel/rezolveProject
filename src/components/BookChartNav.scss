@import '../variables';

.nav-container {
  display: flex;
  margin-top: 20px;
}

.nav-link {
  font-size: 18px;
  color: $nav_font;
}

.nav-link-active {
  font-size: 18px;
  font-weight: bold;
  color: $active;
}

.nav-element-separator {
  font-size: 18px;
  font-weight: bold;
  color: $nav_font;
}
