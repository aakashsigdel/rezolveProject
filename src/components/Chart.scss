@import '../variables';

.chart-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 30px;
}

.chart-label {
  margin-top: 10px;
  color: $active;
  font-size: 18px;
  font-weight: bold;
}
