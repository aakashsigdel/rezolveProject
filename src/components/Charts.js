import React from 'react'

const Charts = () => null

export default Charts
