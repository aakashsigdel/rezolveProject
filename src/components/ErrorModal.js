import React from 'react'
import { Link } from 'react-router'
import { connect } from 'react-redux'

import './ErrorModal.scss'

const ErrorModal = props =>
  <div className="error-container">
    <div className="error-heading">
      <div className="error"> Error </div>
      <div className="error-cross"> âœ• </div>
    </div>
    <div className="error-message">
      {props.message}
    </div>
    <div className="error-buttons">
      <Link to="/">
        <button className="nope" onClick={props.onCancel}>Cancel</button>
    </Link>
      <button className="ok" onClick={props.onReload}>Reload</button>
    </div>
  </div>

export default ErrorModal
