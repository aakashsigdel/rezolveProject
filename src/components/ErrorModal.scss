.error-container {
  position: absolute;
  height: 200px;
  color: #FFF;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background: #D45659;
  width: 500px;
  box-shadow: 0px 0px 10px 0px #352F3C;
  left: 60px;
  top: 60px;
}
.error-heading {
  font-size: 18px;
  padding: 0px 10px 0px 10px;
  align-items: center;
  flex: 1;
  display: flex;
  justify-content: space-between;
  border-bottom: 1px solid #C04C4F; 
}
.error-message {
  flex: 1.5;
  padding: 0px 10px 0px 10px;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 12px;
}
.error-buttons {
  padding-right: 10px;
  flex: 1.3;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  background: white;
}
.nope {
  width: 100px;
  height: 30px;
  border-style: none;
  border-radius: 5px;
  background: #BFBFBF;
  color: white;
  font-size: 18px;
}
.ok {
  font-size: 18px;
  margin: 0px 0px 0px 20px;
  color: white;
  width: 100px;
  height: 30px;
  border-style: none;
  border-radius: 5px;
  background: #DB585C;
}

