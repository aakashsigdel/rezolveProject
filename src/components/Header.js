import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router'
import { NavItem, Button, Nav, Navbar, FormGroup, FormControl } from 'react-bootstrap'
import { pick } from 'ramda'

import { searchChangeText } from '../actions/search'
import { searchBooks } from '../actions/books'
import { searchWillSearch } from '../actions/search'

import './Header.scss'

export const  handleSearch = (props) => {
  props.dispatch(searchBooks(props.search.searchText))
  props.dispatch(searchWillSearch(props.search.searchText))
}

export const Header = (props) =>
  <Navbar>
    <Navbar.Header>
      <Navbar.Brand>
        <a href="#">Rezolve Books</a>
      </Navbar.Brand>
      <Navbar.Toggle />
    </Navbar.Header>
      <Navbar.Form pullLeft>
        <FormGroup>
          <FormControl
            type="text"
            placeholder="Search"
            onChange={e => props.dispatch(searchChangeText(e.target.value))}
          />
        </FormGroup>
        {' '}
        <Link to={"/search/" + props.search.searchText}>
          <Button onClick={() => handleSearch(props)}>
            Search
          </Button>
        </Link>
      </Navbar.Form>
      <div className="pull-right user-container">
        <div className="profile-picture" />
        <div className="user-description">
          <div className="user-name">
            John Doe
          </div>
          <div className="books-number">
            {props.mybooks.books.length + ' books'}
          </div>
        </div>
      </div>
  </Navbar>


export default connect(pick(['search', 'mybooks']))(Header)
