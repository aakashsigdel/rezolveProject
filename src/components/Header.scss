@import "../variables";

.user-container {
  display: flex;
  flex-direction: row;
  margin-right: 30px;
  margin-top: 5px;
}
.profile-picture {
  width: 40px;
  height: 40px;
  border-radius: 20px;
  background-color: $placeholder;
}

.user-description {
  margin-left: 20px;
}

.user-name {
  font-weight: bold;
}

.books-number {
  color: $quantity;
}
