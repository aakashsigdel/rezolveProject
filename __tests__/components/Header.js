import React from 'react'
import { Header } from '../../src/components/Header'

import renderer from 'react-test-renderer'

it('should render <Header /> correctly', () => {
  const tree = renderer.create(
    <Header />
  ).toJSON()

  expect(tree).toMatchSnapshot()
})
