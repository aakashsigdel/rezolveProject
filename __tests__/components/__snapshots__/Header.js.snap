exports[`test should render <Header /> correctly 1`] = `
<nav
  className="navbar navbar-default">
  <div
    className="container">
    <div
      className="navbar-header">
      <a
        className="navbar-brand"
        href="#">
        Rezolve Books
      </a>
      <button
        className="navbar-toggle collapsed"
        onClick={[Function]}
        type="button">
        <span
          className="sr-only">
          Toggle navigation
        </span>
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
      </button>
    </div>
    <div
      className="navbar-form navbar-left">
      <div
        className="form-group">
        <input
          className="form-control"
          id={undefined}
          placeholder="Search"
          type="text" />
      </div>
       
      <button
        className="btn btn-default"
        disabled={false}
        type="submit">
        Search
      </button>
    </div>
    <div
      className="pull-right user-container">
      <div
        className="profile-picture" />
      <div
        className="user-description">
        <div
          className="user-name">
          John Doe
        </div>
        <div
          className="books-number">
          10 books
        </div>
      </div>
    </div>
  </div>
</nav>
`;
